morphia/src/test/java/org/mongodb/morphia/TestQuery.java

```diff
@@ -369,37 +391,67 @@ public void testElemMatchQueryCanBeNegated() {
 
         Mapper mapper = getMorphia().getMapper();
 
-        List<Key<PhotoWithKeywords>> keys = getDs().find(PhotoWithKeywords.class)
+        validate(pwk1, pwk2, pwk3, pwk4, mapper, getDs().find(PhotoWithKeywords.class)
```

Comment:
validate and validate2 are not testing what the test method says it is: that elemMatch queries can be negated. Only validate3 is doing that.  So put the validate/validate2 assertions in a separate test.